### PR TITLE
refactor: cssinjs for Rate

### PR DESCRIPTION
**Pull request by PeachScript**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

#34087 

### 💡 Background and solution

Migrate to cssinjs for Rate component.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Rewrite Rate style with cssinjs. |
| 🇨🇳 Chinese | 使用 cssinjs 重写 Rate 样式。 |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/rate/style/index.tsx [hunk 1]
```diff
@@ -1,5 +1,149 @@
...
+
+const genRateStarStyle = ({ prefixCls, iconPrefixCls, token }: GenRateArgs): CSSObject => ({
+  [`.${prefixCls}-star`]: {
+    position: 'relative',
+    display: 'inline-block',
+    color: 'inherit',
+    cursor: 'pointer',
+
+    '&:not(:last-child)': {
+      marginInlineEnd: 8,
```
Comment (zombieJ) [inline components/rate/style/index.tsx:32]: 用 `token.marginXXX` 代替

File: components/rate/style/index.tsx [hunk 2]
```diff
@@ -1,5 +1,149 @@
...
+    },
+
+    '> div': {
+      transition: 'all 0.3s, outline 0s',
```
Comment (zombieJ) [inline components/rate/style/index.tsx:36]: `0.3s` 用 `token.duration` 代替

Comment (zombieJ) [reply components/rate/style/index.tsx:36]: 下同

File: components/rate/style/index.tsx [hunk 3]
```diff
@@ -1,5 +1,149 @@
...
+
+      [`.${iconPrefixCls}`]: {
+        verticalAlign: 'middle',
+      },
+    },
+
+    '&-first': {
+      position: 'absolute',
+      top: 0,
+      left: 0,
```
Comment (zombieJ) [inline components/rate/style/index.tsx:65]: 用 `insetInlineStart` 代替 `left`

Comment (zombieJ) [reply components/rate/style/index.tsx:65]: 这样 rtl 就只要写个 `direction: rtl` 就会自动切换了

**Comment by WilferCiro** [comment]
Hi, what about add an example where the icon color changes? I test with and is ok
<Rate character="好" allowHalf style={{color: "red"}} />

**Pull request merged**